test/ranges.jl

```diff
@@ -2112,3 +2112,16 @@ end
 @test length(range(1, 100, length=big(100)^100)) == big(100)^100
 @test length(range(big(1), big(100)^100, length=big(100)^100)) == big(100)^100
 @test length(0 * (1:big(100)^100)) == big(100)^100
+
+@testset "issue #41784" begin
+    # test `in` when step equals 0
+    # test for Int
+    x = abs(rand(Int))
```

Comment:
Why the `abs` here? Shouldn't
```suggestion
    x = rand(Int)
```
work as well?